internal/expr/expr_test.go

```diff
@@ -716,7 +716,7 @@ func (s *ExprSuite) TestQueryError(c *C) {
 		query:       "SELECT street FROM t WHERE y = $Person.name",
 		prepareArgs: []any{outerP},
 		queryArgs:   []any{shadowedP},
-		err:         "invalid input parameter: type expr_test.Person not passed as a parameter, have expr_test.Person",
+		err:         `invalid input parameter: type "expr_test.Person" not passed as a parameter (have "expr_test.Person")`,
```

Comment:
I find this error a little bit confusing because it does not tell the user the type was shadowed, something that we know. If it is doable as part of this or another PR, can we change the error message?

Reply:
So in this case the test is a little misleading. I would expect that when this error was normally thrown, shadowed type and the original would be from different packages which would make this error message a lot clearer. Unfortunately its a bit more complicated to simulate that in the test.

You are right in its current form the message is misleading. I will change it to be more explict about the issue, but I won't go so far as to add a specific case for the package *and* the type name being the same. We could change it but then this would be a error for a subcase of a subcase, the first of which we would already expect to be pretty rare (that of the type name being the same but the package name different). My instinct would be to change this if we find out that there is a way people could encounter this error that's more common that the contrived example in the tests.